### ISSUE TITLE
trabalho

### ISSUE THREAD

**Issue by RonaldoBarroncas** [WanderlanAlbuquerque/machine-Learning#2]
`

![Image](https://github.com/user-attachments/assets/7cbd672d-3bb0-4dee-8786-97bdd0d4e0b5)
![Image](https://github.com/user-attachments/assets/6e02de46-e313-413f-8ad3-5299cb656e8a)

from sklearn.metrics import confusion_matrix, accuracy_score, precision_score, recall_score, f1_score
import matplotlib.pyplot as plt
import numpy as np

# Valores reais (exemplo)
y_true = [1, 0, 1, 0, 1, 1, 0, 0, 1, 1, 0, 0, 1, 1, 0, 0, 1, 0, 1, 0]

# Valores preditivos (exemplo)
y_pred = [1, 0, 1, 0, 1, 0, 1, 0, 1, 1, 0, 0, 1, 0, 0, 0, 1, 0, 1, 0]

# Criar a matriz de confusão
matrix = confusion_matrix(y_true, y_pred)

# Converter os valores da matriz para porcentagens
matrix_percent = (matrix / np.sum(matrix)) * 100

# Cálculo das métricas
acuracia = accuracy_score(y_true, y_pred) * 100
precisao = precision_score(y_true, y_pred) * 100
sensibilidade = recall_score(y_true, y_pred) * 100
f1score = f1_score(y_true, y_pred) * 100

# Exibir as métricas no console
print(f"Acurácia: {acuracia:.1f}%")
print(f"Precisão: {precisao:.1f}%")
print(f"Sensibilidade: {sensibilidade:.1f}%")
print(f"F1 Score: {f1score:.1f}%")

# Gráfico de pizza para a matriz de confusão
confusion_labels = ['True Negative (TN)', 'False Positive (FP)', 'False Negative (FN)', 'True Positive (TP)']
confusion_values = [matrix[0, 0], matrix[0, 1], matrix[1, 0], matrix[1, 1]]
confusion_percent = [(value / np.sum(matrix)) * 100 for value in confusion_values]

plt.figure(figsize=(8, 8))
plt.pie(confusion_percent, labels=confusion_labels, autopct=lambda p: f'{p:.1f}%', colors=['green', 'orange', 'red', 'blue'])
plt.title('Distribuição da Matriz de Confusão (%)')
plt.show()

# Gráfico de pizza para as métricas de desempenho
metrics_labels = ['Acurácia', 'Precisão', 'Sensibilidade', 'F1 Score']
metrics_values = [acuracia, precisao, sensibilidade, f1score]

plt.figure(figsize=(8, 8))
plt.pie(metrics_values, labels=metrics_labels, autopct=lambda p: f'{p:.1f}%', colors=['blue', 'orange', 'green', 'red'])
plt.title('Métricas de Desempenho (%)')
plt.show()

**Comment by RonaldoBarroncas** [comment]
ronaldo jose simoes barroncas
